sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jdbc/JdbcPropertiesBeanPostProcessor.java

```diff
@@ -0,0 +1,136 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.core.env.Environment;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.StringUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.MYSQL_PLUGIN_CLASS_NAME;
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.POSTGRES_PLUGIN_CLASS_NAME;
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.PROPERTY_MYSQL_AUTHENTICATION_PLUGINS;
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.PROPERTY_MYSQL_DEFAULT_AUTHENTICATION_PLUGIN;
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.PROPERTY_MYSQL_SSL_MODE;
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.PROPERTY_MYSQL_USE_SSL;
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.PROPERTY_POSTGRESQL_AUTHENTICATION_PLUGIN_CLASSNAME;
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.PROPERTY_POSTGRESQL_SSL_MODE;
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.VALUE_MYSQL_SSL_MODE;
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.VALUE_MYSQL_USE_SSL;
+import static com.azure.spring.cloud.autoconfigure.jdbc.JdbcConnectionStringPropertyConstants.VALUE_POSTGRESQL_SSL_MODE;
+import static com.azure.spring.cloud.core.implementation.util.AzurePropertiesUtils.copyPropertiesIgnoreNull;
+
+/**
+ */
+public class JdbcPropertiesBeanPostProcessor implements BeanPostProcessor, EnvironmentAware {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JdbcPropertiesBeanPostProcessor.class);
+
+    private final AzureGlobalProperties azureGlobalProperties;
+
+    private Environment environment;
+
+    public JdbcPropertiesBeanPostProcessor(AzureGlobalProperties azureGlobalProperties) {
+        this.azureGlobalProperties = azureGlobalProperties;
+    }
+
+    private static final Map<String, String> POSTGRES_ENHANCED_PROPERTIES = new TreeMap<>();
+    private static final Map<String, String> MYSQL_ENHANCED_PROPERTIES = new TreeMap<>();
+    static final Map<DatabaseType, Map<String, String>> ENHANCED_PROPERTIES = new HashMap<>();
+
+    static {
+        POSTGRES_ENHANCED_PROPERTIES.put(PROPERTY_POSTGRESQL_AUTHENTICATION_PLUGIN_CLASSNAME, POSTGRES_PLUGIN_CLASS_NAME);
```

Comment:
can we move these code to the `DatabaseType` to make it more cohesive?

Reply:
updated